nsv13/code/modules/overmap/fighters/modules/ftl_drive.dm

```diff
@@ -28,22 +28,27 @@
 		return INITIALIZE_HINT_LATELOAD
 
 /obj/item/fighter_component/ftl/LateInitialize()
-	set waitfor = FALSE
 	var/obj/structure/overmap/ourfighter = get_overmap()
 	anchored_to = ourfighter.get_overmap()
 
-/obj/item/fighter_component/ftl/tier2
-	name = "class III torch drive"
-	desc = "A micro jump drive with an expanded range."
-	max_range = 200
+/obj/item/fighter_component/ftl/on_install(obj/structure/overmap/target)
+	. = ..()
+	var/obj/structure/overmap/OM = loc
+	OM.ftl_drive = src
+
+/obj/item/fighter_component/ftl/remove_from(obj/structure/overmap/target, due_to_damage)
+	var/obj/structure/overmap/OM = loc
+	. = ..()
+	if(. && istype(OM) && OM.ftl_drive == src)
+		OM.ftl_drive = null
 
 /obj/item/fighter_component/ftl/proc/jump(datum/star_system/target_system, force=FALSE)
 	var/obj/structure/overmap/linked = loc
 	if(!linked)
 		to_chat(usr, "<span class='warning'>This drive was not installed correctly.</span>")
 		message_admins("Fighter FTL drive tried to jump but could not find the linked overmap at [ADMIN_VERBOSEJMP(src)]")
 		return FALSE
-	if(!target_system || !(SSmapping.level_trait(loc.z, ZTRAIT_OVERMAP) || SSmapping.level_trait(loc.z, ZTRAIT_RESERVED)))
+	if(!target_system || !SSmapping.level_trait(loc.z, ZTRAIT_OVERMAP) || SSmapping.level_trait(loc.z, ZTRAIT_RESERVED))
```

Comment:
I believe the fighter ftl issue came from some broken boolean logic here. The parenthesis made this a NOR condition where it should have been two separate conditions, as jumping to reserved z levels should fail.

I'm unsure as to why this PR brought this bug to light though, maybe something to do with changed charging/jump times?